components/tabs/tab-mixin.js

```diff
@@ -0,0 +1,161 @@
+
+	static get properties() {
+		return {
+			selected: { type: Boolean, reflect: true },
+			tabIndex: { type: Number },
+		};
+	};
+
+	static styles = css`
+        :host {
+            box-sizing: border-box;
+            display: inline-block;
+            max-width: 200px;
+            outline: none;
+            position: relative;
+            vertical-align: middle;
+        }
+        .d2l-tab-text {
+            margin: 0.5rem;
+            overflow: hidden;
+            padding: 0.1rem;
+            text-overflow: ellipsis;
+            white-space: nowrap;
+        }
+        :host(:first-child) .d2l-tab-text {
+            margin-left: 0;
+        }
+        .d2l-tab-selected-indicator {
+            border-top: 4px solid var(--d2l-color-celestine);
+            border-top-left-radius: 4px;
+            border-top-right-radius: 4px;
+            bottom: 0;
+            display: none;
+            margin: 1px 0.6rem 0 0.6rem;
+            position: absolute;
+            transition: box-shadow 0.2s;
+            width: calc(100% - 1.2rem);
+        }
+        :host(:first-child) .d2l-tab-selected-indicator {
+			margin-left: 0;
+			width: calc(100% - 0.6rem);
+		}
+        :host(:${unsafeCSS(getFocusPseudoClass())}) > .d2l-tab-text {
+            border-radius: 0.3rem;
+            box-shadow: 0 0 0 2px var(--d2l-color-celestine);
+            color: var(--d2l-color-celestine);
+        }
+        :host([aria-selected="true"]:focus) {
+            text-decoration: none;
+        }
+        :host(:hover) {
+            color: var(--d2l-color-celestine);
+            cursor: pointer;
+        }
+        :host([aria-selected="true"]:hover) {
+            color: inherit;
+            cursor: default;
+        }
+		:host([aria-selected="true"]) .d2l-tab-selected-indicator {
+			display: block;
+		}
+    `;
+
+	constructor() {
+		super();
+		this.role = 'tab';
+		this.selected = false;
+		this.tabIndex = -1;
+
+		this._handleResize = this._handleResize.bind(this);
+		this._resizeObserver = null;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.addEventListener('click', this._handleClick);
+		this.addEventListener('keydown', this._handleKeydown);
+		this.addEventListener('keyup', this._handleKeyup);
+
+		this.resizeObserver = new ResizeObserver(() => {
+			this.dispatchEvent(new CustomEvent('d2l-tab-resize', { bubbles: true, composed: true }));
+		});
+		this.resizeObserver.observe(this);
+	}
+
+	disconnectedCallback() {
+		if (this.resizeObserver) {
+			this.resizeObserver.disconnect();
+			this.resizeObserver = null;
+		}
+		removeEventListener('click', this._handleClick);
+		removeEventListener('keydown', this._handleKeydown);
+		removeEventListener('keyup', this._handleKeyup);
+		super.disconnectedCallback();
+	}
+
+	render() {
+		const contentClasses = {
+			'd2l-tab-text': true,
+		};
+
+		return html`
+			<div
+				class="${classMap(contentClasses)}"
+				aria-selected="${this.selected}"
+				role="tab"
```

Comment:
The `role` and `tabIndex` are specified on the host element (constructor above), which is consistent with `d2l-tab-internal`. Having the `role="tab"` elements on the host means that we can ensure they are direct children of `role="tablist"`, which may be required. It also means we want `aria-selected` sprouted (check out `d2l-tab-internal`) on the host.

I'm open the changing it, but I _think_ we probably want to stick with the host, in which case we won't want `aria-selected`, `role`, or `tabindex` on this `div`. Either way, these need to be specified on one one or the other.